### PR TITLE
LIME-349: Replace mocked data in Home page with real

### LINKED ISSUE #349: Replace mocked data in Home page with real one
Need to replace all the mocked data in Home page with real one

### PR DESCRIPTION
**Pull request by ivanbilobrov07**

### CODE REVIEW

File: frontend/src/bundles/overview/components/chart-goal-progress/helpers/generate-monthly-datasets.helper.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { type WorkoutResponseDto } from '~/bundles/workouts/types/types.js';
+
+import { type ChartGoalDataset } from '../types/chart-goal-dataset.type.js';
+
+const generateMonthlyDatasets = (
```
Comment (dimapopovych) [inline frontend/src/bundles/overview/components/chart-goal-progress/helpers/generate-monthly-datasets.helper.ts:7]: can we simplify all of these a little with date-fns?

File: frontend/src/bundles/overview/components/chart-goal-progress/helpers/convert-stats-to-percentage.helper.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { type ChartGoalDataset } from '../types/types.js';
+
+const defaultValue = 0;
+
+const convertStatsToPercentage = (
```
Comment (dimapopovych) [inline frontend/src/bundles/overview/components/chart-goal-progress/helpers/convert-stats-to-percentage.helper.ts:5]: so as 100% will be displayed largest by steps/kkal on timeframe?

Comment (ivanbilobrov07) [reply frontend/src/bundles/overview/components/chart-goal-progress/helpers/convert-stats-to-percentage.helper.ts:5]: Yes

File: frontend/src/bundles/overview/components/chart-goal-progress/helpers/generate-monthly-datasets.helper.ts (hunk 1 above)
Comment (ivanbilobrov07) [reply frontend/src/bundles/overview/components/chart-goal-progress/helpers/generate-monthly-datasets.helper.ts:7]: I thought about it, but for the moment cant come up with a better approach

Comment (nikita-remeslov) [reply frontend/src/bundles/overview/components/chart-goal-progress/helpers/generate-monthly-datasets.helper.ts:7]: we have date-fns for it, we dont need new package

**Comment by stpalamar** [comment]
Refactored with date-fns and improved UI moments a little bit

https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/26843161/2f8a14c6-fa5f-4930-99b0-c3201f2044b1

**Comment by stpalamar** [comment]
Handling empty schedule and goals


https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/26843161/71c4bcc4-e304-4e0b-90f3-bc8f7236bb17

**Review by DavidCharif — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**